extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -21,7 +25,37 @@ import com.mapbox.maps.extension.style.utils.toValue
  * @see [The online documentation](https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/#geojson)
  *
  */
-class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
+class GeoJsonSource(
+  builder: Builder
+) : Source(builder.sourceId) {
+
+  var geoJsonParsed = false
+    private set
+  private val onGeoJsonParsedListenerList = mutableListOf<(GeoJsonSource) -> Unit>()
+
+  private constructor(
+    builder: Builder,
+    rawGeoJson: GeoJson?,
+    onGeoJsonParsed: (GeoJsonSource) -> Unit
+  ) : this(builder) {
+    rawGeoJson?.let {
+      onGeoJsonParsedListenerList.add(onGeoJsonParsed)
+      workerHandler.post {
+        val property = it.toPropertyValue()
+        mainHandler.post {
+          setProperty(property)
+          geoJsonParsed = true
+          onGeoJsonParsedListenerList.forEach {
+            it.invoke(this)
+          }
+        }
+      }
+    }
+  }
+
+  fun addOnGeoJsonParsedListener(listener: (GeoJsonSource) -> Unit) {
```

Comment:
shall we have a method to remove listener?

Reply:
it is more of internal logic but I guess we could add remove method

Reply:
Added this function.